tests/test_api.py

```diff
@@ -275,14 +275,24 @@ def test_api_get_build_one_unauth(testclient):
 
 def test_api_get_build_one_auth(testclient):
     testclient.login()
-    response = testclient.get("api/v1/build/1")
-    response.raise_for_status()
+    success = False
+    for _ in range(50):
+        response = testclient.get("api/v1/build/1")
+        response.raise_for_status()
 
-    r = schema.APIGetBuild.parse_obj(response.json())
-    assert r.status == schema.APIStatus.OK
-    assert r.data.id == 1
-    assert r.data.specification.name == "python-flask-env"
-    assert r.data.status == schema.BuildStatus.COMPLETED.value
+        r = schema.APIGetBuild.parse_obj(response.json())
+        assert r.status == schema.APIStatus.OK
+        assert r.data.id == 1
+        assert r.data.specification.name == "python-flask-env"
+        if r.data.status in [
+            schema.BuildStatus.QUEUED.value,
+            schema.BuildStatus.BUILDING.value,
+        ]:
+            time.sleep(10)
+            continue
+        assert r.data.status == schema.BuildStatus.COMPLETED.value
+        success = True
+    assert success
```

Comment:
This and the other test were modified because they failed often, due to checking the build before it's available.